### PR TITLE
Tracker updates

### PR DESCRIPTION
**Pull request by schwers**
This updates trackingEvents. Each commit should have the related tickets in each commit. I combined ones that were related.

:eyeglasses: @ajacksified @drunken-economist

### CODE REVIEW

**Comment by drunken-economist** [comment]
:nail_care: just one change to send non-subreddit listings

**Comment by schwers** [comment]
:haircut:

File: assets/js/trackingEvents.es6.js [hunk 1]
```diff
@@ -138,6 +156,16 @@ function trackingEvents(app) {
         data.target_url = target.url;
         data.target_url_domain = target.domain;
       }
+    } else if (props.ctx.params.subreddit || props.ctx.path === "/") {
+      // Fake subreddit, mark it as a listing
+      data.target_type = 'listing';
+
+      // explicitly check that this is the frontpage
+      if (props.ctx.path === "/") {
```
Comment (ajacksified) [inline assets/js/trackingEvents.es6.js]: single quotes

File: assets/js/trackingEvents.es6.js [hunk 2]
```diff
@@ -138,6 +156,16 @@ function trackingEvents(app) {
         data.target_url = target.url;
         data.target_url_domain = target.domain;
       }
+    } else if (props.ctx.params.subreddit || props.ctx.path === "/") {
```
Comment (ajacksified) [inline assets/js/trackingEvents.es6.js]: check for `ctx.params.multi` so you can set multi as the target_type

**Comment by ajacksified** [comment]
:nail_care:

**Comment by schwers** [comment]
adding some example events for the fakesubreddit / multi.

Frontpage `'/'` looks like:
```
{  
   "endpoint_ts":1452641492927,
   "endpoint_ip":"10.1.18.47",
   "event_ts":1452641492588,
   "uuid":"8da352b3-d6b4-4e01-9aad-ed9827b8325c",
   "payload":{  
      "base_url":"/",
      "utc_offset":-8,
      "app_name":"reddit-mobile-test",
      "language":"en",
      "target_sort":"hot",
      "referrer_domain":"schwers.local:4444",
      "dnt":false,
      "referrer_url":"http://schwers.local:4444/u/krispykrackers/m/programming?count=25&page=1&after=t3_40n9bd&before=",
      "domain":"schwers.local:4444",
      "user_agent":"Mozilla/5.0 (iPhone; CPU iPhone OS 8_0 like Mac OS X) AppleWebKit/600.1.3 (KHTML, like Gecko) Version/8.0 Mobile/12A4345d Safari/600.1.4",
      "listing_name":"frontpage",
      "compact_view":false,
      "loid":"O1zt3LGP138S3YWuk1",
      "loid_created":"2015-12-15T21:59:02.391Z",
      "target_count":25,
      "target_type":"listing"
   },
   "event_type":"cs.screenview",
   "event_topic":"screenview_events"
}
```

Second page of `/r/all` looks like:
```
{  
   "endpoint_ts":1452641614666,
   "endpoint_ip":"10.1.19.216",
   "event_ts":1452641614808,
   "uuid":"2a32a1f0-1ee0-42f8-bfa7-7ef0154407df",
   "payload":{  
      "base_url":"/r/all?count=25&page=1&after=t3_40mr59&before=",
      "utc_offset":-8,
      "app_name":"reddit-mobile-test",
      "language":"en",
      "target_sort":"hot",
      "referrer_domain":"schwers.local:4444",
      "domain":"schwers.local:4444",
      "dnt":false,
      "referrer_url":"http://schwers.local:4444/r/all",
      "target_after":"t3_40mr59",
      "user_agent":"Mozilla/5.0 (iPhone; CPU iPhone OS 8_0 like Mac OS X) AppleWebKit/600.1.3 (KHTML, like Gecko) Version/8.0 Mobile/12A4345d Safari/600.1.4",
      "listing_name":"all",
      "compact_view":false,
      "loid":"O1zt3LGP138S3YWuk1",
      "loid_created":"2015-12-15T21:59:02.391Z",
      "target_count":25,
      "target_type":"listing"
   },
   "event_type":"cs.screenview",
   "event_topic":"screenview_events"
}
```

Question re multi reddits for @drunken-economist 

For multi reddit's I'm not sure exactly what you'd like there. For named/linked multireddits (e.g. `/u/krispykrackers/m/programming`) I'm sending 
```
{  
   "event_topic":"screenview_events",
   "event_type":"cs.screenview",
   "event_ts":1452641804744,
   "uuid":"c9007ecd-c356-4781-8df4-1f224cd0a102",
   "payload":{  
      "language":"en",
      "dnt":false,
      "loid":"O1zt3LGP138S3YWuk1",
      "loid_created":"2015-12-15T21:59:02.391Z",
      "target_sort":"hot",
      "target_count":25,
      "compact_view":false,
      "target_type":"multi",
      "multi_name":"programming",
      "app_name":"reddit-mobile-test",
      "utc_offset":-8,
      "user_agent":"Mozilla/5.0 (iPhone; CPU iPhone OS 8_0 like Mac OS X) AppleWebKit/600.1.3 (KHTML, like Gecko) Version/8.0 Mobile/12A4345d Safari/600.1.4",
      "domain":"schwers.local:4444",
      "base_url":"/u/krispykrackers/m/programming"
   }
}
```

Instead of sending `target_type:'listing', listing_name:'multi'` or `target_type:'listing', listing_name:'programming'` Because that seems like it would help you sepreate it from other multi's and the programming subreddit listing. Should this case be something different?

Relatedly: right now `/r/programming+javascript` looks like:
```
{  
   "event_topic":"screenview_events",
   "event_type":"cs.screenview",
   "event_ts":1452642221062,
   "uuid":"86a8a50b-6ddf-4db3-8c0f-a34044b9f9df",
   "payload":{  
      "language":"en",
      "dnt":false,
      "loid":"O1zt3LGP138S3YWuk1",
      "loid_created":"2015-12-15T21:59:02.391Z",
      "target_sort":"hot",
      "target_count":25,
      "compact_view":false,
      "target_type":"listing",
      "listing_name":"programming+javascript",
      "app_name":"reddit-mobile-test",
      "utc_offset":-8,
      "user_agent":"Mozilla/5.0 (iPhone; CPU iPhone OS 8_0 like Mac OS X) AppleWebKit/600.1.3 (KHTML, like Gecko) Version/8.0 Mobile/12A4345d Safari/600.1.4",
      "domain":"schwers.local:4444",
      "base_url":"/r/programming+javascript"
   }
}
```

**Comment by drunken-economist** [comment]
@schwers For multis, we use `"target_type":"listing"`, `"listing_name":"multi"`. This applies to both named multis and manual multis

**Comment by schwers** [comment]
@drunken-economist Got it, updated the multi check, now they look like:

```
{  
   "event_topic":"screenview_events",
   "event_type":"cs.screenview",
   "event_ts":1452647785606,
   "uuid":"d4e150fc-9bf2-49c4-ad7f-933107cbee5e",
   "payload":{  
      "language":"en",
      "dnt":false,
      "loid":"O1zt3LGP138S3YWuk1",
      "loid_created":"2015-12-15T21:59:02.391Z",
      "target_sort":"hot",
      "target_count":25,
      "compact_view":false,
      "target_type":"listing",
      "listing_name":"multi",
      "app_name":"reddit-mobile-test",
      "utc_offset":-8,
      "user_agent":"Mozilla/5.0 (iPhone; CPU iPhone OS 8_0 like Mac OS X) AppleWebKit/600.1.3 (KHTML, like Gecko) Version/8.0 Mobile/12A4345d Safari/600.1.4",
      "domain":"schwers.local:4444",
      "base_url":"/u/krispykrackers/m/programming"
   }
}
```

and 
```
{  
      "event_topic":"screenview_events",
      "event_type":"cs.screenview",
      "event_ts":1452647502305,
      "uuid":"7fbb4331-9ca2-40e8-bb33-fa2fdad77852",
      "payload":{  
         "language":"en",
         "dnt":false,
         "loid":"O1zt3LGP138S3YWuk1",
         "loid_created":"2015-12-15T21:59:02.391Z",
         "target_sort":"hot",
         "target_count":25,
         "compact_view":false,
         "target_type":"listing",
         "listing_name":"multi",
         "app_name":"reddit-mobile-test",
         "utc_offset":-8,
         "user_agent":"Mozilla/5.0 (iPhone; CPU iPhone OS 8_0 like Mac OS X) AppleWebKit/600.1.3 (KHTML, like Gecko) Version/8.0 Mobile/12A4345d Safari/600.1.4",
         "domain":"schwers.local:4444",
         "base_url":"/r/nottheonion+onion"
      }
   }
```

**Comment by schwers** [comment]
:haircut:

File: assets/js/trackingEvents.es6.js [hunk 3]
```diff
@@ -138,6 +161,23 @@ function trackingEvents(app) {
...
+      // Fake subreddit, mark it as a listing
+      data.target_type = 'listing';
+
+      // explicitly check that this is the frontpage
+      if (props.ctx.path === '/') {
+        data.listing_name = 'frontpage';
+      } else if (props.ctx.params.subreddit) {
+        const subreddit = props.ctx.params.subreddit;
+        if (subreddit.indexOf('+') !== -1) {
+          data.listing_name = 'multi';
```
Comment (ajacksified) [inline assets/js/trackingEvents.es6.js]: is this technically a `multi`? (@drunken-economist)

**Comment by ajacksified** [comment]
Code looks fine, pending @drunken-economist's approval for data (:pseudofish:)

File: assets/js/trackingEvents.es6.js (hunk 3 above)
Comment (drunken-economist) [inline assets/js/trackingEvents.es6.js]: For the purposes of our events, we're calling /r/foo+bar a "multi".

**Comment by schwers** [comment]
There's only english right now and we don't get back a language preference on the account afaik.

**Comment by drunken-economist** [comment]
Just went through QA on the event format with this, lgtm. :fish:

**Comment by ajacksified** [comment]
:fish:

**Pull request merged**